### DISCUSSION TITLE
Feature Request: Displaying the file size in Repository Listings

### DISCUSSION THREAD

**Discussion by arizvisa** [community/community#109229] [Repositories]
### Select Topic Area

Product Feedback

### Body

As per the topic, I think it'd be pretty handy to display the file sizes alongside the commit information for each listed item. My specific use case is when browsing other user repositories before deciding on whether a local clone should be performed, or to locate large files full of source code that is probably worth reading (or downloading).

If GH's backend is based on `git`, it's available as the `-l` parameter to `git-ls-tree`. Afaict, though, it's not indexed, and you'd have to explicitly fetch the BLOB size for each TREE associated with the selected commit.

I've been using GH for a pretty long time, and I can't believe I never noticed that there's no file size until now. Perhaps that's a good counter-argument for avoiding the feature entirely. However, despite its subtlety, I still think it's worthwhile and I couldn't have been the first to have desired this.

Apologies in advance if this has been requested before, I searched the "Repositories" tag and didn't see any title that referenced "file size".

**Comment by stevexyz** [comment]
Clone a repository to see the files size as it seems suggested elsewhere is an awful solution. Agree that is a needed feature. Even if not showed immediately there should be a configuration or a button or something else to investigate the sizes!
